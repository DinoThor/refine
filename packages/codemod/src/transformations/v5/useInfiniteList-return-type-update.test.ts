import { describe, it, expect } from "@jest/globals";
import transform from "./useInfiniteList-return-type-update";
import jscodeshift, { type JSCodeshift } from "jscodeshift";

const transformCode = (source: string) => {
  const j: JSCodeshift = jscodeshift.withParser("tsx");

  return transform(
    { source, path: "test.tsx" },
    {
      jscodeshift: j,
      j,
      stats: () => {},
      report: () => {},
    },
    {},
  );
};

describe("useInfiniteList-return-type-update", () => {
  it("should transform basic useInfiniteList destructuring", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          data,
          error,
          hasNextPage,
          isLoading,
          fetchNextPage,
          isFetchingNextPage,
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should handle renamed destructured properties", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          data: posts,
          isLoading: loading,
          hasNextPage,
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should preserve overtime and other non-query/result properties", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          data,
          isLoading,
          overtime,
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should handle only query properties", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          isLoading,
          error,
          fetchNextPage,
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should handle only result properties", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          data,
          hasNextPage,
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should not transform non-destructured usage", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const queryResult = useInfiniteList({
          resource: "posts"
        });

        return <div>{queryResult.data}</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });

  it("should handle TypeScript generic usage", () => {
    const source = `
      import { useInfiniteList } from "@refinedev/core";

      interface IPost {
        id: number;
        title: string;
      }

      const MyComponent = () => {
        const {
          data,
          hasNextPage,
          isLoading,
        } = useInfiniteList<IPost>({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    `;

    expect(transformCode(source)).toMatchSnapshot();
  });
});
