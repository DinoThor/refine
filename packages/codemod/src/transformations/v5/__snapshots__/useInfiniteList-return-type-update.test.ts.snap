// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`useInfiniteList-return-type-update should handle TypeScript generic usage 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      interface IPost {
        id: number;
        title: string;
      }

      const MyComponent = () => {
        const {
          query: {
            isLoading
          },

          result: {
            data,
            hasNextPage
          }
        } = useInfiniteList<IPost>({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should handle only query properties 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          query: {
            isLoading,
            error,
            fetchNextPage
          }
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should handle only result properties 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          result: {
            data,
            hasNextPage
          }
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should handle renamed destructured properties 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          query: {
            isLoading: loading
          },

          result: {
            data: posts,
            hasNextPage
          }
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should not transform non-destructured usage 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const queryResult = useInfiniteList({
          resource: "posts"
        });

        return <div>{queryResult.data}</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should preserve overtime and other non-query/result properties 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          query: {
            isLoading
          },

          result: {
            data
          },

          overtime
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;

exports[`useInfiniteList-return-type-update should transform basic useInfiniteList destructuring 1`] = `
"
      import { useInfiniteList } from "@refinedev/core";

      const MyComponent = () => {
        const {
          query: {
            error,
            isLoading,
            fetchNextPage,
            isFetchingNextPage
          },

          result: {
            data,
            hasNextPage
          }
        } = useInfiniteList({
          resource: "posts"
        });

        return <div>Content</div>;
      };
    "
`;
